src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub GetDirectories_Directory_SearchOption()
+            Dim TestBase As New FileIOTestBase
+            Dim DirectoryList As ReadOnlyCollection(Of String) = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+            Assert.True(DirectoryList.Count = 0)
+            For i As Integer = 0 To 5
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
+            Next
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+            Assert.True(DirectoryList.Count = 6)
+            For i As Integer = 0 To 5
+                Assert.Equal(DirectoryList(i), IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
+            Next
+            IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory0", $"NewSubSubDirectory"))
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+            Assert.True(DirectoryList.Count = 6)
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchAllSubDirectories)
+            Assert.True(DirectoryList.Count = 7)
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub GetDirectories_Directory_SearchOption_Wildcards()
+            Dim TestBase As New FileIOTestBase
+            Dim DirectoryList As ReadOnlyCollection(Of String) = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*0", "*1")
+            Assert.True(DirectoryList.Count = 0)
+            For i As Integer = 0 To 5
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
+            Next
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*0", "*1")
+            Assert.True(DirectoryList.Count = 2)
+            For i As Integer = 0 To 1
+                Assert.Equal(DirectoryList(i), IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
+            Next
+            IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory0", $"NewSubSubDirectory0"))
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*0")
+            Assert.True(DirectoryList.Count = 1)
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchAllSubDirectories, "*0")
+            Assert.True(DirectoryList.Count = 2)
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub GetDirectoryInfo_Directory()
+            Dim StartTime As Date = Now
+            Threading.Thread.Sleep(1000)
+
+            Dim TestBase As New FileIOTestBase
```

Comment:
why we are creating subDirectories here ?

Reply:
There are two search options one that just searches current directory and one that descends down the tree and I need to test that both work correctly.